src/python/library/requirements/requirements_http.txt

```diff
@@ -25,6 +25,6 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 aiohttp>=3.8.1,<4.0.0
-geventhttpclient>=1.4.4,<=2.0.2
+geventhttpclient>=2.0.11
```

Comment:
quick question: is there a version between > 2.0.2 and < 2.0.11 that can run on Python 3.12 but does not have the issue that you are facing?

maybe we should set a max bound, i.e. `<2.1.0`, to avoid accidental breakages on newer releases.

Reply:
No. 2.0.11 is the minimum version to support python3.12 and the issue was introduced in 2.0.4.

Reply:
I agree to set a upper bound. For geventhttpclient we have `2.0.11`, `2.0.12`, `2.1.1`, `2.2.0`, `2.2.1`, `2.3.0`, `2.3.1`.

Reply:
@rmccorm4 Do you think we should fix the version to 2.0.11 or set a upper bound, e.g. 2.1.0?

Reply:
I think upper bound is fine following the convention of our other dependencies

Reply:
Updated